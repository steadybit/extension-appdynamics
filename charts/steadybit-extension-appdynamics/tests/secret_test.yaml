templates:
  - secret.yaml
tests:
  - it: no secret with existing secret
    set:
      appdynamics:
        accessToken: null
        apiClientSecret: null
        existingSecret: existing-secret
    asserts:
      - hasDocuments:
          count: 0
  - it: manifest should match snapshot with apiClientSecret
    set:
      appdynamics:
        apiClientSecret: 111-222-333
        existingSecret: null
    asserts:
      - matchSnapshot: {}
  - it: manifest should match snapshot with accessToken
    set:
      appdynamics:
        accessToken: 111-222-333
        existingSecret: null
    asserts:
      - matchSnapshot: {}
  - it: manifest should match snapshot with both apiClientSecret and accessToken
    set:
      appdynamics:
        accessToken: 111-222-333
        apiClientSecret: 111-222-333
        existingSecret: null
    asserts:
      - matchSnapshot: {}
  - it: no secret without any secrets
    set:
      appdynamics:
        accessToken: null
        apiClientSecret: null
        existingSecret: null
    asserts:
      - hasDocuments:
          count: 0
  - it: should render nothing with default values
    asserts:
      - hasDocuments:
          count: 0

