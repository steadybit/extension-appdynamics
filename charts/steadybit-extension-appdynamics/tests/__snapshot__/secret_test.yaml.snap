manifest should match snapshot with accessToken:
  1: |
    apiVersion: v1
    data:
      accessToken: MTExLTIyMi0zMzM=
    kind: Secret
    metadata:
      labels: null
      name: steadybit-extension-appdynamics
      namespace: NAMESPACE
    type: Opaque
manifest should match snapshot with apiClientSecret:
  1: |
    apiVersion: v1
    data:
      apiClientSecret: MTExLTIyMi0zMzM=
    kind: Secret
    metadata:
      labels: null
      name: steadybit-extension-appdynamics
      namespace: NAMESPACE
    type: Opaque
manifest should match snapshot with both apiClientSecret and accessToken:
  1: |
    apiVersion: v1
    data:
      accessToken: MTExLTIyMi0zMzM=
      apiClientSecret: MTExLTIyMi0zMzM=
    kind: Secret
    metadata:
      labels: null
      name: steadybit-extension-appdynamics
      namespace: NAMESPACE
    type: Opaque
